packages/headless/src/api/platform-client.ts

```diff
@@ -127,3 +122,41 @@ export function platformUrl<E extends PlatformEnvironment = 'prod'>(options?: {
 
   return `https://platform${urlEnv}${urlRegion}.cloud.coveo.com`;
 }
+
+function buildDefaultRequestOptions(
+  options: PlatformClientCallOptions
+): PlatformRequestOptions {
+  const {
+    url,
+    method,
+    requestParams,
+    contentType,
+    accessToken,
+    signal,
+  } = options;
+  const body = encodeBody(requestParams, contentType);
+
+  return {
+    url,
+    method,
+    headers: {
+      'Content-Type': contentType,
+      Authorization: `Bearer ${accessToken}`,
+      ...options.headers,
+    },
+    body,
+    signal,
+  };
+}
+
+function encodeBody(body: unknown, contentType: HTTPContentType) {
+  if (contentType === 'application/json') {
+    return JSON.stringify(body);
+  }
+
+  if (contentType === 'application/x-www-form-urlencoded') {
+    return canBeFormUrlEncoded(body) ? encodeAsFormUrl(body) : '';
+  }
+
+  return '';
+}
```

Comment:
This function is the key change of the PR.

Open question: when something cannot be correctly encoded, do you prefer to return an empty string or to throw an error?

Reply:
To be more pragmatic, I think I would have `'application/json'` and `not sure what to do here` as `JSON.stringify(body)` as the last fallback.

Reply:
Good suggestion, I will make the change.